qa/L0_metrics/cpu_metrics_test.py

```diff
@@ -0,0 +1,187 @@
+                    sequential_dupes = 0
+
+                # For future observability on dupe frequency to tune the tolerance
+                if sequential_dupes > max_sequential_dupes:
+                    max_sequential_dupes = sequential_dupes
+
+                self.assertLess(sequential_dupes, dupe_value_tolerance)
+                prev_value = value
+
+            print(
+                f"Max sequential duplicate values found for {metric}: {max_sequential_dupes}"
+            )
+
+    def _collect_metrics(self, observed_metrics, interval_secs=1):
+        """
+        Collects metrics at provided 'interval_secs' and stores them in the
+        provided 'observed_metrics' dictionary for postprocessing.
+        """
+        # Give the test and server some time to begin processing requests
+        # before beginning observation loop.
+        time.sleep(1)
+
+        while not self.inference_completed.is_set():
+            util_value, used_memory_value, _ = get_metrics()
+            observed_metrics[CPU_UTILIZATION].append(util_value)
+            observed_metrics[CPU_USED_MEMORY].append(used_memory_value)
+            time.sleep(interval_secs)
+
+    def test_cpu_metrics_during_inference(self):
+        with httpclient.InferenceServerClient(
+            url="localhost:8000", concurrency=10
+        ) as client:
+            # Start a thread to collect metrics asynchronously while inferences are
+            # executing, store them in a dictionary for postprocessing validation.
+            observed_metrics = defaultdict(list)
+            metrics_thread = threading.Thread(
+                target=self._collect_metrics, args=(observed_metrics,)
+            )
+            metrics_thread.start()
+
+            # Fire off many asynchronous inference requests to keep server
+            # busy while monitoring the CPU metrics. Ideal target is about
+            # 20-30 seconds of inference to get a good number of metric samples.
+            async_requests = []
+            for _ in range(2000):
+                async_requests.append(
+                    client.async_infer(
+                        model_name=self.model_name,
+                        inputs=self.inputs,
+                    )
+                )
+
+            # Wait for all inference requests to complete
+            for async_request in async_requests:
+                async_request.get_result()
+
+            # Set the event to indicate that inference is completed
+            self.inference_completed.set()
+
+            # Wait for the metrics thread to complete
+            metrics_thread.join()
+
+        self._validate_metric_variance(observed_metrics)
+
+    def test_cpu_metrics_ranges(self):
+        # Test some simple sanity checks on the expected ranges of values
+        # for the CPU related metrics.
+        utilization, used_memory, total_memory = get_metrics()
+        self.assertGreaterEqual(utilization, 0)
+        self.assertLessEqual(utilization, 1.0)
```

Comment:
Also just a reminder that Python can support
```python
self.assertTrue(0 <= utilization <= 1.0)
```
Totally not necessary to change though.